examples/CVAE_example/client.py

```diff
@@ -0,0 +1,75 @@
+from fl4health.tasks.autoencoder_trainer import CVAETrainer
+from fl4health.utils.load_data import load_mnist_data
+from fl4health.utils.metrics import Metric
+from fl4health.utils.sampler import DirichletLabelBasedSampler
+
+
+class CondAutoEncoderClient(CVAETrainer, BasicClient):
+    def __init__(self, data_path: Path, metrics: Sequence[Metric], DEVICE: torch.device, condition: str):
+        CVAETrainer.__init__(self, condition)
+        BasicClient.__init__(self, data_path, metrics, DEVICE)
+
+    def get_data_loaders(self, config: Config) -> Tuple[DataLoader, DataLoader]:
+        batch_size = self.narrow_config_type(config, "batch_size", int)
+        sampler = DirichletLabelBasedSampler(list(range(10)), sample_percentage=0.75, beta=100)
+        sampler.set_seed(42)
+        transform = transforms.Compose([transforms.ToTensor(), transforms.Lambda(torch.flatten)])
+
+        train_loader, val_loader, _ = self.prepare_input(
+            load_data=load_mnist_data,
+            data_path=self.data_path,
+            batch_size=batch_size,
+            sampler=sampler,
+            transform=transform,
+        )
+
+        return train_loader, val_loader
+
+    def get_criterion(self, config: Config) -> _Loss:
+        # The base_loss is the loss function used for comparing the original and generated image pixels.
+        # In this example, data is in binary scale, therefore binary cross entropy is used.
+        # In self.loss(), the base_loss is added to the KL divergence loss.
+        base_loss = torch.nn.BCELoss(reduction="sum")
+        latent_dim = self.narrow_config_type(config, "latent_dim", int)
+        return self.loss(latent_dim, base_loss)
+
+    def get_optimizer(self, config: Config) -> Optimizer:
+        return torch.optim.Adam(self.model.parameters(), lr=0.001)
+
+    def get_model(self, config: Config) -> nn.Module:
+        latent_dim = self.narrow_config_type(config, "latent_dim", int)
+        num_conditions = self.narrow_config_type(config, "num_conditions", int)
+        encoder = MnistConditionalEncoder(input_size=784, num_conditions=num_conditions, latent_dim=latent_dim)
+        decoder = MnistConditionalDecoder(latent_dim=latent_dim, num_conditions=num_conditions, output_size=784)
+        return ConditionalVAE(
+            AutoEncoderType.CONDITIONAL_VAE, num_conditions=num_conditions, encoder=encoder, decoder=decoder
+        )
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="FL Client Main")
+    parser.add_argument("--dataset_path", action="store", type=str, help="Path to the local dataset")
+    parser.add_argument("--condition", action="store", type=str, help="Client ID or 'label' used for CVAE")
+    args = parser.parse_args()
+
+    DEVICE = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+    data_path = Path(args.dataset_path)
+    client = CondAutoEncoderClient(data_path, [], DEVICE, args.condition)
```

Comment:
I think for now passing the client number is fine, but if we were to use this in practice we wouldn't want the clients to have to coordinate their client numbers for a FL training of a CVAE. It may be good to add a ticket to clickup to explore automatically assigning the condition in the server. I am not sure 100 percent how it would work, but if reserve the first round to sample all clients, we can map the cid of each client to a condition number which could be passed in subsequent rounds to encode the input to the CVAE for a given client.